sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -97,7 +97,7 @@ private[execution] object HashedRelation {
         new UnifiedMemoryManager(
           new SparkConf().set(MEMORY_OFFHEAP_ENABLED.key, "false"),
           Long.MaxValue,
-          Long.MaxValue,
+          Long.MaxValue / 2,
```

Comment:
shall we add an assert in `UnifiedMemoryManager` to make sure execution memory is > 0?

Reply:
That's fine by me. I presume it's not strictly an error to specify 0 execution memory, but also can't think of a situation where it's valid.